code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,696 @@
+	dat += "Track Length: [selection.song_length/10] seconds<br><br>"
+	dat += "<i>More songs can be unlocked by earning more IEV points</i><br>"
+	dat += "<br>DJ's Soundboard:<b><br>"
+	dat +="<div class='statusDisplay'><div style='text-align:center'>"
+	dat += "<A href='?src=\ref[src];action=horn'>Air Horn</A>  "
+	dat += "<A href='?src=\ref[src];action=alert'>Station Alert</A>  "
+	dat += "<A href='?src=\ref[src];action=siren'>Warning Siren</A>  "
+	dat += "<A href='?src=\ref[src];action=honk'>Honk</A><br>"
+	dat += "<A href='?src=\ref[src];action=pump'>Shotgun Pump</A>"
+	dat += "<A href='?src=\ref[src];action=pop'>Gunshot</A>"
+	dat += "<A href='?src=\ref[src];action=saber'>Esword</A>"
+	dat += "<A href='?src=\ref[src];action=harm'>Harm Alarm</A>"
+	var/datum/browser/popup = new(user, "vending", "Radiance Dance Machine - Mark IV", 400, 350)
+	popup.set_content(dat.Join())
+	popup.open()
+
+
+/obj/machinery/disco/Topic(href, href_list)
+	if(..())
+		return
+	add_fingerprint(usr)
+	switch(href_list["action"])
+		if("toggle")
+			if (!src || QDELETED(src))
+				return
+			if(!active)
+				if(stop > world.time)
+					to_chat(usr, "<span class='warning'>Error: The device is still resetting from the last activation, it will be ready again in [round((stop-world.time)/10)] seconds.</span>")
+					playsound(src, 'sound/misc/compiler-failure.ogg', 50, 1)
+					return
+				active = TRUE
+				icon_state = "disco1"
+				dance_setup()
+				START_PROCESSING(SSobj, src)
+				lights_spin()
+				src.updateUsrDialog()
+			else if(active)
+				active = FALSE
+				STOP_PROCESSING(SSobj, src)
+				icon_state = "disco0"
+				dance_over()
+				stop = world.time + 300
+				src.updateUsrDialog()
+		if("select")
+			if(active)
+				to_chat(usr, "<span class='warning'>Error: You cannot change the song until the current one is over.</span>")
+				return
+			check_GBP()
+			select_name = input(usr, "Choose your song", "Track:") as null|anything in available
+			if (!src || QDELETED(src))
+				return
+			for(var/datum/track/S in songs)
+				if(select_name == S.song_name)
+					selection = S
+					break
+			src.updateUsrDialog()
+		if("horn")
+			deejay('sound/items/AirHorn2.ogg')
+		if("alert")
+			deejay('sound/misc/notice1.ogg')
+		if("siren")
+			deejay('sound/machines/engine_alert1.ogg')
+		if("honk")
+			deejay('sound/items/bikehorn.ogg')
+		if("pump")
+			deejay('sound/weapons/shotgunpump.ogg')
+		if("pop")
+			deejay('sound/weapons/Gunshot3.ogg')
+		if("saber")
+			deejay('sound/weapons/saberon.ogg')
+		if("harm")
+			deejay('sound/AI/harmalarm.ogg')
+
+/obj/machinery/disco/proc/deejay(var/S)
+	if (!src || QDELETED(src) || !active || charge < 5)
+		to_chat(usr, "<span class='warning'>The device is not able to play more DJ sounds at this time.</span>")
+		return
+	charge -= 5
+	playsound(src, S,300,1)
+
+/obj/machinery/disco/proc/check_GBP()
+	var/point_total = 0
+	available.Cut()
+	for(var/obj/machinery/engi_points_manager/EPM in machines)
```

Comment:
This shouldn't be dependent on the point machine until the point system is actually merged.